### ISSUE TITLE
Some UUIDs Are Not Properly Handled By The Mock Server

### ISSUE THREAD

**Issue by janneronkko** [cannatag/ldap3#894]
The initial problem I had was described in https://github.com/cannatag/ldap3/issues/797

That issue was solved in commit 6fc1917f39141078ce36e48cade95c1f737d5bfa I think, i.e. in version 3.8

There still is some cases when the mock server does not handle ObjectGUIDs properly; for example, ObjectGUID values `1f3e2261-b35c-4065-98e3-3c243f695446` and `5cc91d87-138a-4894-967d-538e7d860020` are converted into bytes when read through LDAP.

The following code demonstrates the issue (and contains the two mentioed UUIDs that don't work):

```python
from uuid import uuid4

import ldap3


def main():
    for guid in (
        '86f66df5-9b0e-4f7d-a6ef-1b897469dcaa',
        '927dc025-c1ad-4fec-bf7b-3e071be62705',
        '3bf75f7e-374e-4bc9-bf1a-2df7fb89d66a',
        '9208a155-c3de-487e-8af7-87e399b5fbc8',
        '1f3e2261-b35c-4065-98e3-3c243f695446',
        '5cc91d87-138a-4894-967d-538e7d860020',
    ):
        check_guid(guid)

    while False:
        check_guid(str(uuid4()))


def check_guid(guid):
        server = ldap3.Server(
            'my_fake_server',
            get_info=ldap3.OFFLINE_AD_2012_R2,
        )
        connection = ldap3.Connection(
            server,
            user='cn=my_user,ou=test,o=lab',
            password='my_password',
            client_strategy=ldap3.MOCK_SYNC,
        )
        connection.bind()

        rv = connection.add(
            dn='cn=user1,ou=test,o=lab',
            object_class=[
                'person',
                'top',
            ],
            attributes={
                'objectGUID': guid,
            },
        )

        connection.search(
            search_base='cn=user1,ou=test,o=lab',
            search_filter='(objectClass=person)',
            search_scope=ldap3.SUBTREE,
            attributes=['objectGUID'],
        )

        assert len(connection.entries) == 1, f'User {dn} not found {connection.entries}'
        object_guid = connection.entries[0].objectGUID
        if isinstance(object_guid.value, str):
            print(f'{object_guid.value} OK')
        else:
            print(f'{guid} -> {object_guid.value} ERROR')


if __name__ == '__main__':
    main()
```

You can quite easily find more non-working UUIDs by changing `while False:` -> `while True` and you will get something like this as an output:

```{86f66df5-9b0e-4f7d-a6ef-1b897469dcaa} OK
{927dc025-c1ad-4fec-bf7b-3e071be62705} OK
{3bf75f7e-374e-4bc9-bf1a-2df7fb89d66a} OK
{9208a155-c3de-487e-8af7-87e399b5fbc8} OK
1f3e2261-b35c-4065-98e3-3c243f695446 -> b'a">\x1f\\5C\xb3e@\x98\xe3<$?iTF' ERROR
5cc91d87-138a-4894-967d-538e7d860020 -> b'\x87\x1d\xc9\\5C\x8a\x13\x94H\x96}S\x8e}\x86\x00 ' ERROR

{b3f459f1-59fc-4cd5-a2cd-290c2932d439} OK
{15ffae45-acd9-4e07-9198-d7aee1b0552a} OK
{63611ec2-1fe6-4608-902b-14729a37dc8e} OK
{d7c5bbb3-48a2-44ec-8f87-8cc1917d3edc} OK
{d4bd0f0b-dc73-4fcb-9f3f-3c6577df1528} OK
{1ef601fe-2fde-400c-a79b-2077357bb809} OK
bd6ebe8e-8684-4a9d-9639-3015a1175c44 -> b'\x8e\xben\xbd\x84\x86\x9dJ\x9690\x15\xa1\x17\\5CD' ERROR
{ec219924-81c9-4704-96ef-66be5ff2cc8e} OK
{d9f2dde4-d919-458d-8359-5878d2e02a5e} OK
{a3476250-b828-48a0-979c-69594e905913} OK
{4ea5491d-20f2-4ee9-b4d1-7f1d13856f4f} OK
{db2c06a8-ca97-41f2-927c-5af0ea1f66eb} OK
{b3daf7b9-a3b3-4fc0-820b-ac53a175b169} OK
{ff3b9467-cc0d-4380-8d04-68fee0b2ade4} OK
```

I have worked that around by using the following function for creating ObjectGUIDs in my tests:
```python
def create_object_guid():
    tries = 0

    object_guid = None
    while not isinstance(object_guid, str):
        assert tries < 5
        tries += 1

        server = ldap3.Server(
            'server',
            get_info=ldap3.OFFLINE_AD_2012_R2,
        )
        connection = ldap3.Connection(
            server,
            user='cn=user,ou=ou,o=o',
            password='my_password',
            client_strategy=ldap3.MOCK_SYNC,
        )
        connection.bind()

        dn = 'cn=new-user,ou=ou,o=o',

        rv = connection.add(
            dn=dn,
            object_class=['person', 'top'],
            attributes={'objectGUID': str(uuid.uuid4())},
        )

        connection.search(
            search_base=dn,
            search_filter='(objectClass=person)',
            search_scope=ldap3.SUBTREE,
            attributes=['objectGUID'],
        )

        assert len(connection.entries)
        object_guid = connection.entries[0].objectGUID.value

    return object_guid

```

**Comment by cannatag** [comment]
Hi, can you check the code in dev? It should work now.

**Comment by janneronkko** [comment]
Thanks, it does work now for me.

I guess the fix is commit 08ff83621322c4a68455712cd3bae02fc084be36? The change looks like it could have an effect and reverting that one did cause the bug to appear again.

**Comment by cannatag** [comment]
Fixed in ldap3 v2.9

**Comment by krizex** [comment]
This fix is incorrect, please see my comments https://github.com/cannatag/ldap3/issues/1000#issuecomment-963846873
